### PR TITLE
Add payment secret feature to Bolt 11 test vectors

### LINKED ISSUE #897: New BOLT 11 Test cases are missing payment secret feature
https://github.com/lightningnetwork/lightning-rfc/commit/42bd71d49c68075f54a31558af67af8c9892a08f added payment secret fields to invoice tests, but didn't add the payment_secret feature flag to them, causing them to be invalid according to the old rules and largely nonsensical as they don't have a feature set for a field they use. We should probably add the payment secret feature flag to them.

### PR DESCRIPTION
**Pull request by t-bast**
Bolt 11 invoices must contain a `payment_secret`, which means that the `features` field must set the `payment_secret` feature (and its dependency, `var_onion_optin`).

Fixes #897

### CODE REVIEW

**Review by TheBlueMatt — COMMENTED** [review]
At least the "Signature is not recoverable." and "Invalid sub-millisatoshi precision." failure cases also need a feature flag, likely others as well but those are just the two that our implementation complains about as missing feature.

**Review by TheBlueMatt — APPROVED** [review]
The changes here LGTM, verified they pass our own tests, though the error cases could use the feature flag as well.

**Comment by t-bast** [comment]
> At least the "Signature is not recoverable." and "Invalid sub-millisatoshi precision." failure cases also need a feature flag, likely others as well but those are just the two that our implementation complains about as missing feature.

Thanks, I wondered where that would be verified in each implementation, in eclair this is verified last, but there's no reason other implementations do the same. I'll update all failure cases that make sense.

File: 11-payment-encoding.md [hunk 1]
```diff
@@ -463,16 +475,17 @@ Breakdown:
...
-> lnbc20m1pvjluezsp5zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zygspp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsfpp3qjmp7lwpagxun9pygexvgpjdc4jdj85fr9yq20q82gphp2nflc7jtzrcazrra7wwgzxqc8u7754cdlpfrmccae92qgzqvzq2ps8pqqqqqqpqqqqq9qqqvpeuqafqxu92d8lr6fvg0r5gv0heeeqgcrqlnm6jhphu9y00rrhy4grqszsvpcgpy9qqqqqqgqqqqq7qqzqg042ea62q6wnxh909rfuu4x2amxc59mj99mrhr32kvqhytrm04us8edg0syy9n0ukgam0ud20jcxwskphv3rzpnengjsf8m3w9u5w8cqzrhtg2
+> lnbc20m1pvjluezsp5zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zygspp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsfpp3qjmp7lwpagxun9pygexvgpjdc4jdj85fr9yq20q82gphp2nflc7jtzrcazrra7wwgzxqc8u7754cdlpfrmccae92qgzqvzq2ps8pqqqqqqpqqqqq9qqqvpeuqafqxu92d8lr6fvg0r5gv0heeeqgcrqlnm6jhphu9y00rrhy4grqszsvpcgpy9qqqqqqgqqqqq7qqzq9qrsgqdfjcdk6w3ak5pca9hwfwfh63zrrz06wwfya0ydlzpgzxkn5xagsqz7x9j4jwe7yj7vaf2k9lqsdk45kts2fd0fkr28am0u4w95tt2nsq76cqw0
```
[the remark this answers is not in the document]
Comment (t-bast) [reply 11-payment-encoding.md:488]: I changed to the friendlier short_channel_id notation that uses blockheight in https://github.com/lightningnetwork/lightning-rfc/pull/898/commits/45222fb396c91d25f91760f0b37fd5a90c4e3541

**Comment by t-bast** [comment]
As you pointed out, only the last two failure test vectors need to be updated, the others have a completely invalid format which should always be checked before the actual contents. I updated them in https://github.com/lightningnetwork/lightning-rfc/pull/898/commits/c32d06a7ac50a9b29bbc8dd18b3c725edae316fc

**Comment by t-bast** [comment]
Merging as per yesterday's spec meeting.

**Pull request merged**